### PR TITLE
Add registered customer edit test

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Adds a logic to test attempts to edit a customer. Adds a new tests checking if an error is thrown when attempting to change customer email to an email of already existing customer. Extracted from https://github.com/PrestaShop/PrestaShop/pull/27769.
| Type?             | new feature
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Check that tests complete OK.
| Fixed ticket?     | 
| Related PRs       | 
| Sponsor company   |

### CODE REVIEW

**Comment by Hlavtox** [comment]
Hello @PrestaShop/qa-automation, can you check and merge, please? 👍

File: tests/Integration/Behaviour/Features/Context/CustomerManagerFeatureContext.php
Comment (Progi1984) [inline tests/Integration/Behaviour/Features/Context/CustomerManagerFeatureContext.php]:
```suggestion
    public function attemptToEditCustomerUsingCommand(string $customerReference, TableNode $table): void
```

**Review by Progi1984 — COMMENTED** [review]
A small comment

File: tests/Integration/Behaviour/Features/Context/CustomerManagerFeatureContext.php
Comment (Hlavtox) [reply tests/Integration/Behaviour/Features/Context/CustomerManagerFeatureContext.php]: Addressed boss ✅

**Review by kpodemski — APPROVED** [review]

**Comment by boubkerbribri** [comment]
> Hello @PrestaShop/qa-automation, can you check and merge, please? +1

Hello @Hlavtox , 
@Progi1984 did a review but FYI, for integration tests, you need committers approvals not qa-automation.

**Review by zuk3975 — APPROVED** [review]

**Comment by zuk3975** [comment]
thanks @Hlavtox

**Pull request merged**
